cpp/src/arrow/util/thread_pool.cc

```diff
@@ -58,6 +60,19 @@ SerialExecutor::~SerialExecutor() = default;
 
 Status SerialExecutor::SpawnReal(TaskHints hints, FnOnce<void()> task,
                                  StopToken stop_token, StopCallback&& stop_callback) {
+#ifdef ARROW_WITH_OPENTELEMETRY
+  // Wrap the task to propagate a parent tracing span to it
+  struct {
+    void operator()() {
+      auto scope = ::arrow::internal::tracing::GetTracer()->WithActiveSpan(activeSpan);
+      std::move(func)();
+    }
+    FnOnce<void()> func;
+    opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> activeSpan;
+  } wrapper{std::forward<FnOnce<void()>>(task),
```

Comment:
Nit, but the struct declarations are subtly different here and in future.cc, maybe we can keep them consistent? (I think the style here is more common in the codebase.)

Reply:
I've changed both of them into the same style as another task wrapper I saw a bit lower in `future.cc` (in `RunOrScheduleCallback`